typed-racket-test/succeed/unsafe-require.rkt

```diff
@@ -0,0 +1,32 @@
+#lang racket/base
+
+;; Test unsafe require forms
+
+(module a racket/base
+  (struct foo (x y))
+  (define a-foo (foo 1 2))
+  (provide (struct-out foo) a-foo))
+
+(module b typed/racket
```

Comment:
Note: adding a line `(provide (struct-out foo))` here raises an exception:
```
struct-out: multiple bindings (foo4 and foo3) for structure-type identifier
  at: foo4
  in: (struct-out foo)
```

Instead, you need to provide the identifiers individually. But be careful, because a typed module can't import them straight away. Adding two lines to `module c`:
```
(require (submod ".." b))
(foo-x (foo "yo" "lo"))
```

Raises some require/typed errors:
```
Type Checker: missing type for identifier;
 consider using `require/typed' to import it
  identifier: foo-x
```